src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -12,6 +12,14 @@
             [slingshot.slingshot :as sling]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Constants
+
+(def error-message-404
+  (str "Error: This master is running Puppet 4, and the URL format has changed. "
+       "This request was likely made with a Puppet 3 Agent, as it used an invalid "
+       "URL format. Please upgrade your agents to Puppet 4 and try again."))
```

Comment:
@nfagerlund As it stands right now with the way I've implemented this, any route that does not have the new configurable master or CA prefix will fall through to this handler and see the 404 error message. I can change that if necessary, however.